frame/salary/src/lib.rs

```diff
@@ -0,0 +1,512 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Make periodic payment to members of a ranked collective according to rank.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![recursion_limit = "128"]
+
+use codec::{Decode, Encode, FullCodec, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_arithmetic::traits::{Saturating, Zero};
+use sp_core::TypedGet;
+use sp_runtime::{
+	traits::{AtLeast32BitUnsigned, Convert},
+	Perbill,
+};
+use sp_std::{fmt::Debug, marker::PhantomData, prelude::*};
+
+use frame_support::{
+	dispatch::DispatchResultWithPostInfo,
+	ensure,
+	traits::{tokens::fungible, RankedMembers},
+	RuntimeDebug,
+};
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+pub mod weights;
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+/// Payroll cycle.
+pub type Cycle = u32;
+
+/// Status for making a payment via the `Pay::pay` trait function.
+#[derive(Encode, Decode, Eq, PartialEq, Clone, TypeInfo, MaxEncodedLen, RuntimeDebug)]
+pub enum PaymentStatus {
+	/// Payment is in progress. Nothing to report yet.
+	InProgress,
+	/// Payment status is unknowable. It will never be reported successful or failed.
+	Unknown,
+	/// Payment happened successfully.
+	Success,
+	/// Payment failed. It may safely be retried.
+	Failure,
+}
+
+// Can be implemented by Pot pallet with a fixed Currency impl, but can also be implemented with
+// XCM/MultiAsset and made generic over assets.
+pub trait Pay {
```

Comment:
By "Pot pallet", do you mean some pallet with a pot of funds from which payments will be made? In many cases the Treasury but possibly others of course.

```suggestion
/// Can be implemented by Pot pallet with a fixed Currency impl, but can also be implemented with
/// XCM/MultiAsset and made generic over assets.
pub trait Pay {
```